modules/ROOT/pages/file/file-read.adoc

```diff
@@ -5,36 +5,149 @@ endif::[]
 :keywords: file, connector, read
 
 
+The File connector can read a file at any point in the flow, unlike in Mule 3, where the
+transport can only read files as a result of inbound endpoint polling.
 
-The File connector can read a file at any point in the flow, unlike in Mule 3, where the transport can only read files as a result of inbound endpoint polling.
+The only required parameter for the `read` operation is `path`, which is the *relative path* of the file
+you wish to read. This path is relative to the `working directory` defined in the configuration
+referenced by the `config-ref` parameter. To see an example of creating a configuration check
+xref:file/file-connector.adoc#connection_settings[this page]. If no configuration is referenced,
+the working directory defaults to the value of the `user.home` system property. If the system
+property is not set, the connector fails to initialize.
 
-Here is the syntax for reading a file:
+Here is an example for reading a file:
 
 [source,xml,linenums]
 ----
-<file:read path="#[path]"
-  lock="true|false"
-  outputEncoding="UTF-8"
-  outputMimeType="application/csv; headers=false" />
+<file:read config-ref="File_Config" path="filePath"/>
 ----
 
-The operation above reads the file in the given path. It returns a `MuleMessage` with the following attributes:
+The `read` operation returns a `MuleMessage` with the following attributes:
 
 * The file's content as payload.
 * The file's metadata in the message attributes.
 * If the file does not exist, you receive a `FILE:ILLEGAL_PATH` error. Note that the operation does not read directories.
 
+[NOTE]
+===========
+The `read` operation (as other operations that return non-void values) supports target parameters.
+To learn more about them check the xref:4.1@mule-runtime::target-variables.adoc[Mule Runtime Target Variables document].
+===========
+
+== Configuring a relative path
+
+Suppose that you have numerous `read` operations (or any other file operations) in your app and they
+share part of their path, you can extract that base path into a
+xref:file/file-connector.adoc#connection_settings[file configuration]. Imagine that we want to access
+two files whose full paths are: `/Users/Documents/Examples/manual.txt` and `/Users/Documents/Examples/readme.txt`.
+We can define the following configuration:
+
+[source,xml,linenums]
+----
+<file:config name="File_Config" >
+    <file:connection workingDir="/Users/Documents/Examples" />
+</file:config>
+----
+
+And then we can define our `read` operations referencing that configuration like this:
+
+[source,xml,linenums]
+----
+<file:read config-ref="File_Config" path="manual.txt" />
+...
```

Comment:
wrap it inside of a flow and delete the 3 dots, try to avoid invalid elements on examples for easier copy and paste